### PR TITLE
lib/protoparser/graphite: added -graphite.sanitizeMetricName flag

### LINKED ISSUE #6077: Implement sanitize logic for graphite as in other protocols .
### Is your feature request related to a problem? Please describe

As I was in the process of migrating  my old **Graphite** into **VictoriaMetrics**  I started to changes between them in the metrics names 

**In **Graphite****
sending `double..dots..test.testing.name1` displays the metric name as `double.dots.test.testing.name1`
In MictoraMetrics 

**But in VictoraMetrics** I see the following  `double..dots..test.testing.name1`
graphite has this capability built-in
When metrics are processed and stored by Graphite, it automatically sanitizes the metric names to ensure they adhere to its naming conventions. This sanitization process involves normalizing consecutive dots (..) into a single dot (.) in metric names

### Describe the solution you'd like

VictoriaMetrcis already has sanitize logic for various protocols. I think we should implement that for graphite as well.

This behavior is design and helps avoid unintended namespace segmentation caused by typographical errors or inconsistent naming practices. For instance, if a metric inadvertently includes multiple dots between namespace segments (like double..dots..test.testing.name1), Graphite will store this metric as double.dots.test.testing.name1, treating the extra dots as a single delimiter. This

### Describe alternatives you've considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Added flag to sanitize graphite metrics
fixes #6077

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: lib/protoparser/graphite/parser_test.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ func TestUnmarshalMetricAndTagsFailure(t *testing.T) {
 }
 
 func TestUnmarshalMetricAndTagsSuccess(t *testing.T) {
+	*sanitizeMetricName = true
```
Comment (zekker6) [inline lib/protoparser/graphite/parser_test.go:22]: nit: it would be great to save original value of the flag and restore it after the test completion

**Review by zekker6 — APPROVED** [review]
LGTM, see 1 minor comment

File: lib/protoparser/graphite/parser_test.go (hunk 1 above)
Comment (AndrewChubatiuk) [reply lib/protoparser/graphite/parser_test.go:23]: done

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Should we mention somewhere, that sanitizing makes metric labels format compatible with prometheus?

**Review by hagen1778 — APPROVED** [review]
LGTM
see my update at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6489/commits/69f906c10130b201e5f796ce1216c664f037efca

**Comment by valyala** [comment]
@AndrewChubatiuk , @hagen1778 , please see the follow-up commit 74affa3aecd769ef94cc4f50774c1759518d8619

**Pull request merged**